app/models/concerns/hud_reports/incomes.rb

```diff
@@ -203,14 +203,23 @@ module HudReports::Incomes
           {
             'Adults with Missing Income Information' => a_t[:annual_assessment_expected].eq(true).
               and(a_t[:annual_assessment_in_window].eq(true)).
-              and(a_t["income_total_at_#{suffix}"].eq(99).
-                or(a_t["income_total_at_#{suffix}"].eq(nil))),
+              and(a_t["income_from_any_source_at_#{suffix}".to_sym].eq(nil).
+                or(a_t["income_from_any_source_at_#{suffix}"].eq(99)).
+                or(
+                  a_t["income_total_at_#{suffix}"].eq(nil).
+                  and(
+                    # Responses of 8 & 9 are expected to have total income nil.
+                    # These are filtered out to prevent duplicates. They are captured in a different row.
+                    a_t["income_from_any_source_at_#{suffix}"].not_eq(8).
+                    and(a_t["income_from_any_source_at_#{suffix}"].not_eq(9)),
```

Comment:
I think the following would have the same result and might be a bit easier to read
```suggestion
                    a_t["income_from_any_source_at_#{suffix}"].not_in([8, 9]),
```

Reply:
This has been updated in [this commit](https://github.com/greenriver/hmis-warehouse/pull/5000/commits/6d299d090919f6b3640dbb18b822a4c0eae6593c)